raulin0/backup-juggler
# Release/0.2.0

## Description
This pull request includes a series of changes and additions aimed at improving the project configuration, documentation, testing, and CI/CD pipeline. The updates provide enhanced functionality, improved documentation, and automated testing processes. Additionally, it bumps the project version from 0.1.0 to 0.2.0.

## Changes Made
- Bumped the project version from 0.1.0 to 0.2.0 to reflect the latest changes and additions.
- Updated the README file with the latest information and usage instructions.
- Merged pull request #1 from raulin0/feature/backup-feature, which introduced the backup feature.
- Fixed the installation process of CI and Read the Docs documentation dependencies.
- Updated the project configuration, metadata, and documentation URLs to ensure accurate and up-to-date information.
- Added a .readthedocs.yml configuration file to facilitate the generation of documentation using Read the Docs (RTD).
- Updated various documentation files, including the contribution guide, index page, tutorial, installation instructions, and API documentation for Backup Juggler.
- Updated the MkDocs configuration file to reflect changes in the documentation generator tool.
- Implemented additional tests for the Backup Juggler CLI and added a fixture for running CLI tests.
- Added the Backup Juggler CLI functionality to allow users to interact with the project through the command line.
- Implemented unit tests for the backup functionality and added fixtures for temporary paths to ensure reliable testing.
- Implemented the file backup functionality with a progress viewer and introduced a paths model class for managing file paths and directories during the backup process.
- Set up a GitHub Actions pipeline for testing and code coverage, ensuring the project's functionality remains intact and maintaining a certain level of code coverage.

## Motivation and Context
These changes were made to improve the project's overall functionality, maintainability, and documentation. The Backup Juggler project benefits from enhanced configuration, automated testing, improved documentation, and the addition of a CLI for user convenience. By making these improvements and bumping the version, the project becomes more user-friendly, easier to contribute to, and better equipped for ongoing maintenance.

## How to Test
1. Review the updated project version, which has been bumped from 0.1.0 to 0.2.0.
2. Read through the updated README file, which provides the latest information and usage instructions for the project.
3. Confirm the successful merge of pull request #1 from raulin0/feature/backup-feature, which added the backup feature.
4. Verify that the installation process of CI and Read the Docs documentation dependencies has been fixed and works correctly.
5. Check that the project configuration, metadata, and documentation URLs have been updated accurately.
6. Validate the presence and correctness of the .readthedocs.yml configuration file for Read the Docs.
7. Read and familiarize yourself with the updated documentation files, including the contribution guide, index, tutorial, installation instructions, and API documentation for Backup Juggler.
8. Ensure that the MkDocs configuration file reflects any changes made during the update process.
9. Run additional tests for the Backup Juggler CLI functionality and confirm they pass successfully.
10. Interact with the Backup Juggler CLI through the command line and verify that the expected functionality is available and functioning correctly.
11. Execute the unit tests for the backup functionality and confirm that they pass, using the added fixtures for temporary paths.
12. Perform a file backup operation using the updated functionality and confirm that the progress viewer accurately reflects the progress of the backup process.
13. Check that the paths model class correctly manages file paths and directories during the backup operation.
14. Validate the GitHub Actions pipeline by ensuring that tests are executed and code coverage reports are generated successfully.

## Related Issue(s)
N/A

## Checklist


Please review and merge this pull request at your earliest convenience. Feel free to provide any feedback or suggestions for improvement. Thank you!
